consensus/safety-rules/src/safety_rules.rs

```diff
@@ -462,6 +465,37 @@ impl SafetyRules {
         let signature = self.sign(timeout)?;
         Ok(signature)
     }
+
+    fn guarded_sign_commit_vote(
+        &mut self,
+        ledger_info: LedgerInfoWithSignatures,
+        new_ledger_info: LedgerInfo,
+    ) -> Result<Ed25519Signature, Error> {
+        self.signer()?;
+
+        // Verify that ledger_info contains at least 2f + 1 dostinct signatures
+        ledger_info
+            .verify_signatures(&self.epoch_state()?.verifier)
+            .map_err(|error| Error::InvalidQuorumCertificate(error.to_string()))?;
+
+        let old_ledger_info = ledger_info.ledger_info();
```

Comment:
we should check that the block info is not BlockInfo::empty() and also check the executed_state_id, version, epoch_state is the same dummy value as we put in the ordering only state computer.

Reply:
Resolved